app/src/common/shared/org/mozilla/vrbrowser/browser/engine/SessionStore.java

```diff
@@ -64,48 +57,10 @@ private SessionStore() {
     public void setContext(Context context, Bundle aExtras) {
         mContext = context;
 
-        if (mRuntime == null) {
-            // FIXME: Once GeckoView has a prefs API
-            SessionUtils.vrPrefsWorkAround(context, aExtras);
-
-            GeckoRuntimeSettings.Builder runtimeSettingsBuilder = new GeckoRuntimeSettings.Builder();
-            runtimeSettingsBuilder.crashHandler(CrashReporterService.class);
-            runtimeSettingsBuilder.contentBlocking((new ContentBlocking.Settings.Builder()
-                    .antiTracking(ContentBlocking.AntiTracking.AD | ContentBlocking.AntiTracking.SOCIAL| ContentBlocking.AntiTracking.ANALYTIC))
-                    .build());
-            runtimeSettingsBuilder.consoleOutput(SettingsStore.getInstance(context).isConsoleLogsEnabled());
-            runtimeSettingsBuilder.displayDensityOverride(SettingsStore.getInstance(context).getDisplayDensity());
-            runtimeSettingsBuilder.remoteDebuggingEnabled(SettingsStore.getInstance(context).isRemoteDebuggingEnabled());
-            runtimeSettingsBuilder.displayDpiOverride(SettingsStore.getInstance(context).getDisplayDpi());
-            runtimeSettingsBuilder.screenSizeOverride(SettingsStore.getInstance(context).getMaxWindowWidth(),
-                    SettingsStore.getInstance(context).getMaxWindowHeight());
-            runtimeSettingsBuilder.autoplayDefault(SettingsStore.getInstance(mContext).isAutoplayEnabled() ? GeckoRuntimeSettings.AUTOPLAY_DEFAULT_ALLOWED : GeckoRuntimeSettings.AUTOPLAY_DEFAULT_BLOCKED);
-
-            if (SettingsStore.getInstance(context).getTransparentBorderWidth() > 0) {
-                runtimeSettingsBuilder.useMaxScreenDepth(true);
-            }
-
-            if (BuildConfig.DEBUG) {
-                runtimeSettingsBuilder.arguments(new String[] { "-purgecaches" });
-                runtimeSettingsBuilder.debugLogging(true);
-                runtimeSettingsBuilder.aboutConfigEnabled(true);
-            } else {
-                runtimeSettingsBuilder.debugLogging(SettingsStore.getInstance(context).isDebugLogginEnabled());
-            }
-
-            mRuntime = GeckoRuntime.create(context, runtimeSettingsBuilder.build());
-            for (String extension: WEB_EXTENSIONS) {
-                String path = "resource://android/assets/web_extensions/" + extension + "/";
-                mRuntime.registerWebExtension(new WebExtension(path));
-            }
-
-        } else {
-            mRuntime.attachTo(context);
-        }
-    }
+        mRuntime = EngineProvider.INSTANCE.getOrCreateRuntime(context);
 
-    public GeckoRuntime getRuntime() {
-        return mRuntime;
+        // FIXME: Once GeckoView has a prefs API
+        SessionUtils.vrPrefsWorkAround(context, aExtras);
```

Comment:
I think the work around function should be called before the runtime is created.

Reply:
Fixed